modules/dxp/apps/portal-search-tuning/portal-search-tuning-blueprints-admin-web/src/main/resources/META-INF/resources/js/edit_blueprint/ResultListItem.js

```diff
@@ -26,24 +26,25 @@ const RESULTS_DEFAULT_KEYS = [
 	'b_created',
 	'b_modified',
 	'b_author',
-	'b_assetEntryId',
 ];
-const RESULTS_HIDE_KEYS = ['explanation', 'document', 'b_viewURL'];
+const RESULTS_SHOW_KEYS = ['b_assetEntryId', 'id'];
 
 function ResultListItem({item}) {
 	const [collapse, setCollapse] = useState(true);
 
-	const _renderListRow = (property) => (
-		<ClayLayout.Row justify="start" key={`${item.id}_${property}`}>
+	const _renderListRow = (property, value, id) => (
+		<ClayLayout.Row justify="start" key={`${id}_${property}`}>
 			<ClayLayout.Col className="semibold" size={4}>
-				{property}
+				{property.match(/b_[\w_]+/g) ? property.substring(2) : property}
```

Comment:
Can we extract this regex logic into a separate function outside of the component?

It's a specialized type of behavior so it'd be good to distinguish it apart from the component itself. Regex is also pretty difficult to cipher what's exactly being matched so it'll help make it more understandable.

Also a few additional notes:

- This current regex seems to match "b_" anywhere in the string.
![Screen Shot 2021-04-16 at 6 02 51 PM](https://user-images.githubusercontent.com/4496722/115097264-f75d9b00-9edd-11eb-9424-fa5224bed86c.png)
- We can simplify this using `.replace` (See full example below)

I think something like this would make it easier to maintain.

```javascript
const BLUEPRINT_FIELD_PREFIX = "b_";

const blueprintFieldPrefixRegex = new RegExp(`^(${BLUEPRINT_FIELD_PREFIX})`);

function removeBlueprintFieldPrefix(value) {
	return value.replace(blueprintFieldPrefixRegex, "");
}
```

And then using it:

```javascript
removeBlueprintFieldPrefix(property);
```

Reply:
Oooo thank you! That looks way better, I'll add these in!